libc/test/src/stdio/sprintf_test.cpp

```diff
@@ -169,6 +170,52 @@ TEST(LlvmLibcSPrintfTest, IntConv) {
   EXPECT_EQ(written, 20);
   ASSERT_STREQ(buff, "-9223372036854775808"); // ll min
 
+  written = LIBC_NAMESPACE::sprintf(buff, "%w3d", 5807);
```

Comment:
you also need tests that check the behavior of incorrect bit-widths. You need at least one test each for `%w0d`, `%w999d`, and `%w-1d`.